pkg/agent/apiserver/handlers/policyconjunctions/handler.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package policyconjunctions
+
+import (
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	agentquerier "antrea.io/antrea/pkg/agent/querier"
+	cpv1beta "antrea.io/antrea/pkg/apis/controlplane/v1beta2"
+	"antrea.io/antrea/pkg/querier"
+)
+
+// From user shorthand input to cpv1beta1.NetworkPolicyType
+var mapToNetworkPolicyType = map[string]cpv1beta.NetworkPolicyType{
+	"K8SNP": cpv1beta.K8sNetworkPolicy,
+	"ACNP":  cpv1beta.AntreaClusterNetworkPolicy,
+	"ANNP":  cpv1beta.AntreaNetworkPolicy,
+	"ANP":   cpv1beta.AdminNetworkPolicy,
+}
+
+var clusterScopedResources = sets.New[string]("ACNP", "ANP")
+
+// HandleFunc creates a http.HandlerFunc which uses an AgentNetworkPolicyInfoQuerier
+// to query network policy rules in current agent.
+func HandleFunc(aq agentquerier.AgentQuerier) http.HandlerFunc {
```

Comment:
wouldn't it make sense to add this information (list of conj IDs) to the existing `networkpolicy` handler instead of defining a new HTTP endpoint? We can always enrich it if we need to support a few additional query parameters? Likewise, maybe we could use the existing antctl command and have a flag to display conjunction IDs?

Reply:
As of now the response type of the `antctl get networkpolicy` command is basically `controlplane.NetworkPolicy` since it's a shared endpoint between controller and agent. As a result it might be a bit tricky to add a conjunction ID field to the Response type, as for the controller it is this antctl command is a `resourceEndpoint` command https://github.com/antrea-io/antrea/blob/main/pkg/antctl/antctl.go#L227

Reply:
There are some differences already in the Agent command compared to the Controller command, and the handlers are different, but I think I see your point: we still use the same response type for both handlers, and it may not be a good idea to change that.

Is there any information that would be useful for debugging on the Agent-side besides conjunction IDs? I am wondering if we should have a more generic (but still Agent-specific) endpoint / command, not one named specifically "policyconjunctions". For example, would it make sense to also include the ruleID in the response, or anything else?

Reply:
Ummm if you are talking about the ID of the rule stored in agent, usually it does not carry information that we can use during debugging. It is a hash that's computed based on all fields of a rule and not really relatable to any controlplane rule info.

Reply:
I opened #6487 which fixed some old antctl ovs-dump command that would essentially show conjunction ID information (buried in the flows). However, it feels to me that it's still quite different from the command provided in this PR: the ovs-dump returns the datapath state, which, in times of troubleshooting, might return empty flows based on the match keys stored. Wondering if this PR is still warranted @tnqn